File: app/schemas/events.py
from __future__ import annotations

from datetime import datetime
from typing import Optional
from uuid import UUID

from app.schemas.common import BaseSchema


class PlatformEventRead(BaseSchema):
    id: UUID
    business_id: Optional[UUID]
    location_id: Optional[UUID]
    schema_version: int
    event_type: str
    compatibility_event_name: Optional[str]
    entity_type: str
    entity_id: Optional[UUID]
    actor_type: str
    actor_user_id: Optional[UUID]
    actor_membership_id: Optional[UUID]
    trace_id: str
    ip_address: Optional[str]
    user_agent: Optional[str]
    payload: dict
    event_metadata: dict
    error_message: Optional[str]
    occurred_at: datetime

File: web/lib/types/events.ts
export type PlatformEvent = {
  id: string;
  business_id?: string | null;
  location_id?: string | null;
  schema_version: number;
  event_type: string;
  compatibility_event_name?: string | null;
  entity_type: string;
  entity_id?: string | null;
  actor_type: string;
  actor_user_id?: string | null;
  actor_membership_id?: string | null;
  trace_id: string;
  ip_address?: string | null;
  user_agent?: string | null;
  payload: Record<string, unknown>;
  event_metadata: Record<string, unknown>;
  error_message?: string | null;
  occurred_at: string;
};

export type PlatformEventListQuery = {
  locationId?: string | null;
  entityType?: string | null;
  eventType?: string | null;
  limit?: number;
};
